lib/block_view.go

```diff
@@ -2774,6 +2817,47 @@ func (bav *UtxoView) _disconnectNFTBid(
 		currentTxn, txnHash, utxoOpsForTxn[:operationIndex+1], blockHeight)
 }
 
+func (bav *UtxoView) _disconnectAuthorizeDerivedKey(
+	operationType OperationType, currentTxn *MsgBitCloutTxn, txnHash *BlockHash,
+	utxoOpsForTxn []*UtxoOperation, blockHeight uint32) error {
+
+	// Verify that the last operation is a AuthorizeDerivedKey operation
+	if len(utxoOpsForTxn) == 0 {
+		return fmt.Errorf("_disconnectAuthorizeDerivedKey: utxoOperations are missing")
+	}
+	operationIndex := len(utxoOpsForTxn) - 1
+	if utxoOpsForTxn[operationIndex].Type != OperationTypeAuthorizeDerivedKey {
+		return fmt.Errorf("_disconnectAuthorizeDerivedKey: Trying to revert "+
+			"OperationTypeAuthorizeDerivedKey but found type %v",
+			utxoOpsForTxn[operationIndex].Type)
+	}
+	txMeta := currentTxn.TxnMeta.(*AuthorizeDerivedKeyMetadata)
+
+	// Revert the Db to the previous authorize entry from utxoOps
+	authEntry := utxoOpsForTxn[operationIndex].AuthEntry
+	senderPKID := PublicKeyToPKID(currentTxn.PublicKey)
+	derivedPKID := PublicKeyToPKID(txMeta.DerivedPublicKey)
+	err := DBPutDerivedKeyMapping(bav.Handle, senderPKID, derivedPKID, authEntry)
```

Comment:
DB puts / gets should not be used in connect/disconnect methods. Keep them in setMappings helpers and let the flush do all DB writing